ambry-network/src/main/java/com.github.ambry.network/SocketServer.java

```diff
@@ -419,9 +420,8 @@ public void run() {
         List<NetworkReceive> completedReceives = selector.completedReceives();
         for (NetworkReceive networkReceive : completedReceives) {
           String connectionId = networkReceive.getConnectionId();
-          Object buffer = networkReceive.getReceivedBytes().getAndRelease();
-          SocketServerRequest req = new SocketServerRequest(id, connectionId, buffer,
-              Utils.createDataInputStreamFromBuffer(buffer, networkConfig.networkPutRequestShareMemory));
+          ByteBuf buffer = networkReceive.getReceivedBytes().content();
+          SocketServerRequest req = new SocketServerRequest(id, connectionId, buffer);
```

Comment:
who is supposed to release the buffer?

Reply:
in the RequestHandler, we have a for loop to get the request out and after dealing with the request, we release it.